examples/v1beta1/pipelineruns/alpha/pipeline-object-results.yaml

```diff
@@ -12,56 +12,71 @@ spec:
               type: object
               description: The object results
               properties:
-                foo:
+                url:
                   type: string
-                hello:
+                digest:
                   type: string
           steps:
             - name: write-object
               image: bash:latest
               script: |
                 #!/usr/bin/env bash
-                echo -n "{\"foo\":\"bar\",\"hello\":\"world\"}" | tee $(results.object-results.path)
+                echo -n "{\"url\":\"abc.dev/sampler\",\"digest\":\"19f02276bf8dbdd62f069b922f10c65262cc34b710eea26ff928129a736be791\"}" | tee $(results.object-results.path)
       - name: task2
         params:
           - name: whole-object
             value: "$(tasks.task1.results.object-results[*])"
-          - name: object-element
-            value: "$(tasks.task1.results.object-results.hello)"
+          - name: object-url
+            value: "$(tasks.task1.results.object-results.url)"
         taskSpec:
           params:
+            - name: object-url
+              type: string
+              default: "default/url"
             - name: whole-object
               type: object
               properties:
-                foo:
+                url:
                   type: string
-                hello:
+                digest:
                   type: string
               default: {
```

Comment:
Shall we try dropping the default from here?

Reply:
I think it was added to show how to use `default` for object params, any reasons we want to drop it?

Reply:
I would suggest creating a separate section for documenting how to use `default` instead of mixing the functionality check and usage. We can add a separate example similar to how we have an example to show default usage for array params

Doc: https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#default-value
Example: https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/taskruns/array-default.yaml

Reply:
That sounds good to me!